### PR TITLE
batch_system: use lru cache

### LINKED ISSUE #8315: Memory  usage gradually grows up and OOM when running random merge
### What version of TiKV are you using?

Master (8faf2601b981f3ac58b902f8a1c0d24ba41b60b1)

### What operating system and CPU are you using?

k8s, CPU: require 4, limit 16;  memory require 16G, limit 32G

### Steps to reproduce

1. Load some data, generate lot of regions
image
image

2. Enable random merge

3. Wait

### What did you expect?

TiKV runs normally

### What did happened?

TiKV's memory usage gradually grows up and OOM

image

...and it's not used by block cache

image

I don't know the reason but I'm afraid there are some memory leak.

cc @BusyJay

### PR DESCRIPTION
**Pull request by BusyJay**
### What problem does this PR solve?

Issue Number: close #8315  

Problem Summary:

When a mailbox is closed, it maybe kept in cache and never be dropped.
This PR uses lru cache to limit the impact and prevent memory from
growing infinitely.

### Related changes

- PR to update `pingcap/docs`/`pingcap/docs-cn`:
- PR to update `pingcap/tidb-ansible`:
- Need to cherry-pick to the release branch

### Check List 

Tests 

- Unit test
- Integration test

### Release note 
- fix memory leak during scheduling

### CODE REVIEW

**Comment by Little-Wallace** [comment]
It seems that the performance is much worse than before...

**Comment by Little-Wallace** [comment]
I think it is not necessary to adjust lru chain every time calling 'get' from cache. How about sample ten percent?

**Comment by breeswish** [comment]
Our PR benchmark platform is not stable. I recommend not to trust it unless for > 20% differences (and <0.1 p).

File: components/tikv_util/src/lru.rs [hunk 1]
```diff
@@ -0,0 +1,415 @@
...
+            suture(&mut self.head, &mut self.tail);
+        }
+    }
+
+    fn remove_tail(&mut self) -> K {
+        unsafe {
+            let mut record = self.tail.prev;
+            cut_out(record.as_mut());
+
+            let r = Box::from_raw(record.as_ptr());
```
Comment (NingLin-P) [inline components/tikv_util/src/lru.rs:129]: Why not use `ptr::drop_in_place` here?

File: components/tikv_util/src/lru.rs [hunk 2]
```diff
@@ -0,0 +1,415 @@
...
+            r.key.as_ptr().read()
```
Comment (NingLin-P) [inline components/tikv_util/src/lru.rs:130]: Why not use `ptr::drop_in_place` here?

Comment (BusyJay) [reply components/tikv_util/src/lru.rs:130]: Because `key` is moved and returned, then freed outside the function.

**Pull request merged**